caluma/visibilities.py

```diff
@@ -1,31 +1,54 @@
-from graphene.utils.str_converters import to_snake_case
+import inspect
+from functools import wraps
 
 
-class BaseVisibility(object):
-    """Basic visibility classes to be extended by any visibility implementation."""
+def filter_queryset_for(node):
+    """Decorate function to define filtering of queryset of specific node."""
 
-    def get_base_queryset(self, node, queryset, info):
-        """
-        Get base queryset.
+    def decorate(fn):
+        @wraps(fn)
+        def _decorated(self, node, queryset, info):
+            return fn(self, node, queryset, info)
 
-        Override to implement default visiblity filters for all nodes.
-        """
-        return queryset
+        _decorated._filter_queryset_for = node
+        return _decorated
 
-    def get_queryset(self, node, queryset, info):
-        """
-        Get queryset of specific node.
+    return decorate
 
-        Checks for node specific implement by its snake name.
 
-        e.g. for Node WorkItem it will call `get_queryset_for_work_item` if available.
-        """
-        queryset = self.get_base_queryset(node, queryset, info)
+class BaseVisibility(object):
+    """Basic visibility classes to be extended by any visibility implementation.
+
+    In combination with the decorator `@queryset_for` a custom visibility class
```

Comment:
Naming - the decorator is called `filter_queryset_for`, not `queryset_for`

Reply:
well spotted. Changed.